discovery_engine_core/core/src/utils.rs

```diff
@@ -55,3 +55,8 @@ pub(crate) fn nan_safe_f32_cmp_base(a: &f32, b: &f32, nan_min: bool) -> Ordering
 pub(crate) fn nan_safe_f32_cmp(a: &f32, b: &f32) -> Ordering {
     nan_safe_f32_cmp_base(a, b, true)
 }
+
+/// Normalizes `text` to a trimmed lowercase string.
+pub(crate) fn normalize(text: &str) -> String {
```

Comment:
I'll prefer to keep it together with the other code, it does not seems that useful in general at the moment.
If we `trim` before `to_lowercase` we can avoid the last `to_string`.

Reply:
thanks, moved back to `article`.